app/models/manageiq/providers/openstack/cloud_manager/provision/volume_attachment.rb

```diff
@@ -1,10 +1,20 @@
 module ManageIQ::Providers::Openstack::CloudManager::Provision::VolumeAttachment
   def create_requested_volumes(requested_volumes)
-    volumes_attrs_list = [default_volume_attributes]
+    volumes_attrs_list = instance_type.root_disk_size > 0 ? [default_volume_attributes] : []
 
     connection_options = {:service => "volume", :tenant_name => cloud_tenant.try(:name)}
     source.ext_management_system.with_provider_connection(connection_options) do |service|
-      requested_volumes.each do |volume_attrs|
+      requested_volumes.each_with_index do |volume_attrs, index|
+        if instance_type.root_disk_size == 0 # Handle root disk logic only when root_disk_size is 0
+          if index == 0  # The first volume should be the root disk, typically assigned to vda
```

Comment:
This might be better done up above like
```
requested_volumes.first&.merge!(:boot_index => 0, :bootable => true, :imageRef => source.ems_ref) if instance_type.root_disk_size.zero?
```

Reply:
> I don't really like the -1, but if that we there previously, then it's fine. I would think null would be a better choice.

I don't think that is a choice since these volume_attrs are for fog-openstack volume create.  From https://docs.openstack.org/nova/rocky/user/launch-instance-from-volume.html I don't see any mention of -1 so I'm going to drop this part so we can try to get this in.